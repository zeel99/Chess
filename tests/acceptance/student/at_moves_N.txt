-- Setup Knight in each of the 16 slots on board and see all its possible moves.

setup_chess(N, 1, 1)
setup_chess(N, 3, 2)
start_game
moves(1, 1)
reset_game

setup_chess(N, 1, 2)
setup_chess(N, 3, 3)
start_game
moves(1, 2)
reset_game

setup_chess(N, 1, 3)
setup_chess(N, 3, 4)
start_game
moves(1, 3)
reset_game

setup_chess(N, 1, 4)
setup_chess(N, 3, 3)
start_game
moves(1, 4)
reset_game

setup_chess(N, 2, 1)
setup_chess(N, 4, 2)
start_game
moves(2, 1)
reset_game

setup_chess(N, 2, 2)
setup_chess(N, 4, 3)
start_game
moves(2, 2)
reset_game

setup_chess(N, 2, 3)
setup_chess(N, 4, 4)
start_game
moves(2, 3)
reset_game

setup_chess(N, 2, 4)
setup_chess(N, 4, 3)
start_game
moves(2, 4)
reset_game

setup_chess(N, 3, 1)
setup_chess(N, 4, 3)
start_game
moves(3, 1)
reset_game

setup_chess(N, 3, 2)
setup_chess(N, 4, 4)
start_game
moves(3, 2)
reset_game

setup_chess(N, 3, 3)
setup_chess(N, 4, 1)
start_game
moves(3, 3)
reset_game

setup_chess(N, 3, 4)
setup_chess(N, 4, 2)
start_game
moves(3, 4)
reset_game

setup_chess(N, 4, 1)
setup_chess(N, 2, 2)
start_game
moves(4, 1)
reset_game

setup_chess(N, 4, 2)
setup_chess(N, 2, 3)
start_game
moves(4, 2)
reset_game

setup_chess(N, 4, 3)
setup_chess(N, 2, 4)
start_game
moves(4, 3)
reset_game

setup_chess(N, 4, 4)
setup_chess(N, 2, 3)
start_game
moves(4, 4)
reset_game