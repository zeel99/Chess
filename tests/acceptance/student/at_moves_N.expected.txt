  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,1)
  # of chess pieces on board: 1
  Game being Setup...
  N...
  ....
  ....
  ....
->setup_chess(N,3,2)
  # of chess pieces on board: 2
  Game being Setup...
  N...
  ....
  .N..
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  N...
  ....
  .N..
  ....
->moves(1,1)
  # of chess pieces on board: 2
  Game In Progress...
  N...
  ..+.
  .+..
  ....
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,2)
  # of chess pieces on board: 1
  Game being Setup...
  .N..
  ....
  ....
  ....
->setup_chess(N,3,3)
  # of chess pieces on board: 2
  Game being Setup...
  .N..
  ....
  ..N.
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  .N..
  ....
  ..N.
  ....
->moves(1,2)
  # of chess pieces on board: 2
  Game In Progress...
  .N..
  ...+
  +.+.
  ....
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..N.
  ....
  ....
  ....
->setup_chess(N,3,4)
  # of chess pieces on board: 2
  Game being Setup...
  ..N.
  ....
  ...N
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ..N.
  ....
  ...N
  ....
->moves(1,3)
  # of chess pieces on board: 2
  Game In Progress...
  ..N.
  +...
  .+.+
  ....
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,4)
  # of chess pieces on board: 1
  Game being Setup...
  ...N
  ....
  ....
  ....
->setup_chess(N,3,3)
  # of chess pieces on board: 2
  Game being Setup...
  ...N
  ....
  ..N.
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ...N
  ....
  ..N.
  ....
->moves(1,4)
  # of chess pieces on board: 2
  Game In Progress...
  ...N
  .+..
  ..+.
  ....
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,2,1)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  N...
  ....
  ....
->setup_chess(N,4,2)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  N...
  ....
  .N..
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  N...
  ....
  .N..
->moves(2,1)
  # of chess pieces on board: 2
  Game In Progress...
  ..+.
  N...
  ..+.
  .+..
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,2,2)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  .N..
  ....
  ....
->setup_chess(N,4,3)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  .N..
  ....
  ..N.
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  .N..
  ....
  ..N.
->moves(2,2)
  # of chess pieces on board: 2
  Game In Progress...
  ...+
  .N..
  ...+
  +.+.
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,2,3)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ..N.
  ....
  ....
->setup_chess(N,4,4)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ..N.
  ....
  ...N
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ..N.
  ....
  ...N
->moves(2,3)
  # of chess pieces on board: 2
  Game In Progress...
  +...
  ..N.
  +...
  .+.+
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,2,4)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ...N
  ....
  ....
->setup_chess(N,4,3)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ...N
  ....
  ..N.
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ...N
  ....
  ..N.
->moves(2,4)
  # of chess pieces on board: 2
  Game In Progress...
  .+..
  ...N
  .+..
  ..+.
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,3,1)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  N...
  ....
->setup_chess(N,4,3)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ....
  N...
  ..N.
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ....
  N...
  ..N.
->moves(3,1)
  # of chess pieces on board: 2
  Game In Progress...
  .+..
  ..+.
  N...
  ..+.
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,3,2)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  .N..
  ....
->setup_chess(N,4,4)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ....
  .N..
  ...N
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ....
  .N..
  ...N
->moves(3,2)
  # of chess pieces on board: 2
  Game In Progress...
  +.+.
  ...+
  .N..
  ...+
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,3,3)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ..N.
  ....
->setup_chess(N,4,1)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ....
  ..N.
  N...
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ....
  ..N.
  N...
->moves(3,3)
  # of chess pieces on board: 2
  Game In Progress...
  .+.+
  +...
  ..N.
  +...
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,3,4)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ...N
  ....
->setup_chess(N,4,2)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ....
  ...N
  .N..
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ....
  ...N
  .N..
->moves(3,4)
  # of chess pieces on board: 2
  Game In Progress...
  ..+.
  .+..
  ...N
  .+..
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,4,1)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ....
  N...
->setup_chess(N,2,2)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  .N..
  ....
  N...
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  .N..
  ....
  N...
->moves(4,1)
  # of chess pieces on board: 2
  Game In Progress...
  ....
  .+..
  ..+.
  N...
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,4,2)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ....
  .N..
->setup_chess(N,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ..N.
  ....
  .N..
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ..N.
  ....
  .N..
->moves(4,2)
  # of chess pieces on board: 2
  Game In Progress...
  ....
  +.+.
  ...+
  .N..
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,4,3)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ....
  ..N.
->setup_chess(N,2,4)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ...N
  ....
  ..N.
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ...N
  ....
  ..N.
->moves(4,3)
  # of chess pieces on board: 2
  Game In Progress...
  ....
  .+.+
  +...
  ..N.
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,4,4)
  # of chess pieces on board: 1
  Game being Setup...
  ....
  ....
  ....
  ...N
->setup_chess(N,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  ....
  ..N.
  ....
  ...N
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ..N.
  ....
  ...N
->moves(4,4)
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ..+.
  .+..
  ...N
->reset_game
  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
